src/doc/version-switcher.js

```diff
@@ -0,0 +1,269 @@
+    filler.id = "switch-version-filler";
+    document.body.appendChild(filler);
+
+    function changeSidebarTop() {
+        const height = switcherEl.getBoundingClientRect().height;
+        const sidebar = document.querySelector(".sidebar");
+        sidebar.style.top = height + 1 + "px";
+    }
+    setTimeout(() => {
+        const sidebar = window.getComputedStyle(document.querySelector(".sidebar"));
+        filler.style.width = sidebar.width;
+        filler.style.backgroundColor = sidebar.backgroundColor;
+        changeSidebarTop();
+    }, 0); // it'll be computed once it's added in the DOM.
+    window.addEventListener("resize", changeSidebarTop);
+}
+
+function setupStyleFor22(rustdoc_container, switcherEl) {
+    // It's mostly the same as `setupStyleFor32` so we call it and make the extra changes afterward.
+    setupStyleFor32(rustdoc_container, switcherEl, `@media (max-width: 700px) {
+    .sidebar {
+        height: 45px;
+        min-height: 40px;
+        margin: 0;
+        margin-left: -15px;
+        padding: 0 15px;
+        position: static;
+        z-index: 1;
+    }
+}`);
+}
+
+function setupStyleFor21(rustdoc_container, switcherEl) {
+    // It's mostly the same as `setupStyleFor22` so we call it and make the extra changes afterward.
+    document.body.style.padding = "0";
+
+    const css = `.rustdoc {
+    padding: 10px 15px 20px 15px !important;
+}`;
+    addStyle(css);
+
+    function changeSidebarTop() {
+        const height = switcherEl.getBoundingClientRect().height;
+        const sidebar = document.querySelector(".sidebar");
+        sidebar.style.top = height + 1 + "px";
+    }
+    setTimeout(() => {
+        changeSidebarTop();
+    }, 0); // it'll be computed once it's added in the DOM.
+    window.addEventListener("resize", changeSidebarTop);
+}
+
+function getHtmlForSwitcher(isOldVersion, switcher_container) {
+    if (!isOldVersion) {
+        switcher_container.style.color = "#eee";
+        return "You can pick a different version with this dropdown:&nbsp;";
+    }
+
+    switcher_container.style.color = "#e57300";
+
+    addStyle(`#doc-version-switcher svg {
+    width: 1em;
+    height: 1em;
+    fill: currentColor;
+    padding-top: 0.1em;
+}`);
+
+    const warning_img = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512">\
+<path d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 \
+0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 \
+416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 \
+46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 \
+11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-\
+11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"></path></svg>&nbsp;`;
+    return warning_img + "You are seeing an outdated version of this documentation. " +
+        "Click on the dropdown to go to the latest stable version:&nbsp;";
```

Comment:
Do you have a screenshot of what this warning looks like? I can't get it to show with the dev-doc since it's only for old versions.